controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1755,17 +1795,24 @@ func (r *BareMetalHostReconciler) createHostFirmwareComponents(info *reconcileIn
 
 			// Set bmh as owner, this makes sure the resource is deleted when bmh is deleted
 			if err = controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
-				return errors.Wrap(err, "could not set bmh as controller")
+				return errors.Wrap(err, "could not set bmh as controller for hostFirmwareComponents")
 			}
 			if err = r.Create(info.ctx, hfc); err != nil {
 				return errors.Wrap(err, "failure creating hostFirmwareComponents resource")
 			}
 
 			info.log.Info("created new hostFirmwareComponents resource")
-		} else {
-			// Error reading the object
-			return errors.Wrap(err, "could not load hostFirmwareComponents resource")
+			return nil
 		}
+		// Error reading the object
+		return errors.Wrap(err, "could not load hostFirmwareComponents resource")
+	}
+	// Necessary in case the CRD is manually created.
+	if err := controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
+		return errors.Wrap(err, "could not set bmh as controller for hostFirmwareComponents")
+	}
+	if err := r.Update(info.ctx, hfc); err != nil {
```

Comment:
Is it going to get called on every reconciliation?

Reply:
you are probably right, let me do a test  having the Update condition nested with the previous if in case the err is nil

Reply:
Done, I just tested the change.